packages/rolldown/src/utils/compose-js-plugins.ts

```diff
@@ -4,19 +4,14 @@ import { normalizeHook } from './normalize-hook'
 import { isNullish } from './misc'
 import { BuiltinPlugin } from '../plugin/builtin-plugin'
 
-const unsupportedHooks = new Set([
-  'augmentChunkHash',
-  'banner',
-  'generateBundle',
-  'moduleParsed',
-  'onLog',
-  'options',
-  'outputOptions',
-  'renderError',
-  'renderStart',
-  'resolveDynamicImport',
-  'writeBundle',
-  'resolveId',
+// FIXME: Conflict with the `skip` option in `PluginContext#resolve`. Since we can't detect it in advance,
+// we have to bailout all plugins with `resolveId` hook.
+const supportedHooks = new Set([
```

Comment:
Using `supportedHooks` rather than `unsupportedHooks` could avoid accentently unsync the logic between filter and `createComposedPlugin` impl, e.g. 

before we did not specify `footer` in `unsupportedHooks`, this could cause a panic when we try to compose two continuous plugins that use the `footer` hook.